### PR TITLE
KAFKA-17919: enable back the failing testShareGroups test

### PR DESCRIPTION
**Pull request by Yunyung**
JIRA: https://issues.apache.org/jira/browse/KAFKA-17919

I have looped the test 40 times on my local, all pass.

Command:
`I=0; while ./gradlew core:test --tests PlaintextAdminIntegrationTest.testShareGroups --rerun --fail-fast; do (( I=$I+1 )); echo "Completed run: $I"; sleep 1; done`

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by AndrewJSchofield** [comment]
I think I had another failure of this test. Let's be careful here.

**Review by apoorvmittal10 — APPROVED** [review]
Thanks for enabling it back.

**Comment by apoorvmittal10** [comment]
> I think I had another failure of this test. Let's be careful here.

Do you mean after the fix you did, this test is again flaky? Is it the same NOT_COORDINATOR error?

**Comment by AndrewJSchofield** [comment]
Yes, it's the same NOT_COORDINATOR error. I ran it extensively on 2 computers yesterday and it was clean on both. The CI and everyone else's testing also appears clean. However, on my third computer today, I get the same failure. I don't understand why, yet.

**Comment by apoorvmittal10** [comment]
> Yes, it's the same NOT_COORDINATOR error. I ran it extensively on 2 computers yesterday and it was clean on both. The CI and everyone else's testing also appears clean. However, on my third computer today, I get the same failure. I don't understand why, yet.

The other doubt I had was NOT_COORDINATOR is a retriable exception hence should we retry as part of https://github.com/apache/kafka/pull/17645 (ofcourse it was not the right fix for testShareGroups test but not sure why a retriable exception is not retried)

**Comment by AndrewJSchofield** [comment]
The test is still flaky, with the same symptom, but a different reason. The fix in https://github.com/apache/kafka/pull/17656 looks good, but there is more to it. I would say there's a timing problem in initialization.

**Comment by AndrewJSchofield** [comment]
When https://github.com/apache/kafka/pull/17645 is merged, then I think the test will be properly deflaked.

**Comment by Yunyung** [comment]
Thanks for both clarifying and explaining this. Let's wait until #17645  is merged.

**Review by chia7712 — APPROVED** [review]

**Pull request merged**
